src/CR.cc

```diff
@@ -282,7 +281,10 @@ void checkSpanForCRs(detection::Footprint *extras, // Extra spans get added to t
             }
             loc.image() = corr;
 
-            extras->addSpan(y + imageY0, x + imageX0, x + imageX0);
+            std::vector<geom::Span> spanList(extras->getSpans()->begin(),
+                                              extras->getSpans()->end());
+            spanList.push_back(geom::Span(y + imageY0, x + imageX0, x +imageX0));
+            extras->setSpans(std::make_shared<geom::SpanSet>(std::move(spanList)));
```

Comment:
We spoke about this in person, but just to note here for posterity, this behavior is intended such that a spanSet is an immutable object to protect any other code that has a shared pointer to a  SpanSet from predictable behavior if someone modifies the SpanSet member of a Footprint. I will make a note in the documentation on the addSpan member about this, and encourage if multiple spans are to be added it would be better to construct a vector of spans, and construct a new SpanSet once, and set it back with setSpans